### PR TITLE
Distilbert pl

### PR DESCRIPTION
**Pull request by ekmb**
1. DistilBert module added and supported in p&c model
2. Switch between sentencepiece and nemobert tokenizers added to the training scripts for ner and p&c
3. Micro av/accuracy added to the classification report
4. Rearranged configs in ner and p&c to reduce duplication for train/dev

### CODE REVIEW

File: examples/nlp/token_classification/conf/ner_config.yaml [hunk 1]
```diff
@@ -13,34 +13,25 @@ pl:
 model:
   data_dir: ??? # /path/to/data
   class_balancing: null # choose from [null, weighted_loss]. Weighted_loss enables the weighted class balancing of the loss, may be used for handling unbalanced classes
+  max_seq_length: 128
+  pad_label: 'O'
+  ignore_extra_tokens: false
+  ignore_start_end: false
+  overwrite_processed_files: false
+  batch_size: 32
+  num_workers:  2
```
Comment (VahidooX) [inline examples/nlp/token_classification/conf/ner_config.yaml:22]: num_workers, pin_memory, and drop_last, batch size are for data loaders? Shouldn't they be there? Now the data loaders parts are very short. batch size can be different for train and evaluation as evaluation can be higher to speed up that part.

File: examples/nlp/token_classification/conf/punctuation_capitalization_config.yaml [hunk 2]
```diff
@@ -8,39 +8,29 @@ pl:
...
 
 model:
   data_dir: ??? # /path/to/data
-  class_balancing: null # or weighted_loss
+  max_seq_length: 128
+  pad_label: 'O'
+  ignore_extra_tokens: false
+  ignore_start_end: false
+  overwrite_processed_files: false
+  batch_size: 32
```
Comment (VahidooX) [inline examples/nlp/token_classification/conf/punctuation_capitalization_config.yaml:20]: The same here.

File: nemo/collections/nlp/models/token_classification/ner_model.py [hunk 3]
```diff
@@ -51,15 +51,18 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         """
 
         self.data_desc = TokenClassificationDataDesc(
-            data_dir=cfg.data_dir, modes=["train", "test", "dev"], pad_label=cfg.train_ds.pad_label
+            data_dir=cfg.data_dir, modes=["train", "test", "dev"], pad_label=cfg.pad_label
         )
         self.data_dir = cfg.data_dir
+        self.model_cfg = cfg
```
Comment (VahidooX) [inline nemo/collections/nlp/models/token_classification/ner_model.py:57]: cfg or cfg.model?

File: nemo/collections/nlp/models/token_classification/punctuation_capitalization_model.py [hunk 4]
```diff
@@ -98,14 +106,18 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
...
+    def forward(self, input_ids, attention_mask, token_type_ids=None):
         """
         No special modification required for Lightning, define it as you normally would
         in the `nn.Module` in vanilla PyTorch.
+        token_type_ids are not used for DistilBert models
         """
-        hidden_states = self.bert_model(
-            input_ids=input_ids, token_type_ids=token_type_ids, attention_mask=attention_mask
-        )
+        if 'distilbert' in self.model_cfg.language_model.pretrained_model_name:
```
Comment (VahidooX) [inline nemo/collections/nlp/models/token_classification/punctuation_capitalization_model.py:115]: Do we need this? Passing None for token_type_ids isn't OK?

File: nemo/collections/nlp/modules/common/huggingface/huggingface_utils.py [hunk 5]
```diff
@@ -57,6 +60,12 @@ def get_huggingface_lm_model(pretrained_model_name: str, config_file: Optional[s
         'config': BertConfig,
         'model_list': BERT_PRETRAINED_MODEL_ARCHIVE_LIST,
     },
+    'distilbert': {
```
Comment (VahidooX) [inline nemo/collections/nlp/modules/common/huggingface/huggingface_utils.py:63]: We should complete this list as least for BERT for the next release.

Comment (VahidooX) [reply nemo/collections/nlp/modules/common/huggingface/huggingface_utils.py:63]: Lets add it to our TODO list.

File: examples/nlp/token_classification/conf/ner_config.yaml (hunk 1 above)
Comment (ekmb) [reply examples/nlp/token_classification/conf/ner_config.yaml]: fixed

File: examples/nlp/token_classification/conf/punctuation_capitalization_config.yaml (hunk 2 above)
Comment (ekmb) [reply examples/nlp/token_classification/conf/punctuation_capitalization_config.yaml]: fixed

File: nemo/collections/nlp/models/token_classification/punctuation_capitalization_model.py (hunk 4 above)
Comment (ekmb) [reply nemo/collections/nlp/models/token_classification/punctuation_capitalization_model.py]: changed the distilbert module

File: nemo/collections/nlp/modules/common/huggingface/distilbert.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    @classmethod
+    def restore_from(cls, restore_path: str):
+        pass
+
+    @typecheck()
+    def forward(self, **kwargs):
+        # distilBert does not use token_type_ids as the most of the other Bert models
+        if 'token_type_ids' in kwargs:
+            logging.info("DistilBert doesn’t use token_type_ids.")
```
Comment (VahidooX) [inline nemo/collections/nlp/modules/common/huggingface/distilbert.py:44]: It is OK. How about adding another sentence ", and it is ignored."

Comment (ekmb) [reply nemo/collections/nlp/modules/common/huggingface/distilbert.py]: updated
